inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -672,21 +687,32 @@ void RemovePermutationsNHWCToNCHWPass::run() {
             IE_ASSERT(next != nullptr);
 
             if (LayerInfo(next).isConvolution()) {
-                next->input()->setDims(toRemove->input()->getDims());
                 next->input()->setLayout(Layout::NHWC);
-                auto layerBeforePermute = CNNNetPrevLayer(toRemove);
-
                 DataPtr output = nullptr;
-                for (auto before_output : layerBeforePermute->outData) {
-                    if (areEqualDatas(toRemove->input(), before_output)) {
-                        output = before_output;
-                        output->setLayout(Layout::NHWC);
-                        break;
+                if (LayerInfo(toRemove).isPermute()) {
```

Comment:
previously logic was then next or prev is permute, now next can be any of permute and reshape, same with prev. Why do we start handling reshape in between permutes-conv-permute?